packages/opentelemetry-core/src/context/propagation/B3Propagator.ts

```diff
@@ -57,7 +57,13 @@ export class B3Propagator implements HttpTextPropagator {
       // We set the header only if there is an existing sampling decision.
       // Otherwise we will omit it => Absent.
       if (spanContext.traceFlags !== undefined) {
-        setter(carrier, X_B3_SAMPLED, Number(spanContext.traceFlags));
+        setter(
+          carrier,
+          X_B3_SAMPLED,
+          (TraceFlags.SAMPLED & spanContext.traceFlags) === TraceFlags.SAMPLED
```

Comment:
shouldn't the `TraceFlags.SAMPLED` become `'1'` then instead of `0x1 << 0` ?

Reply:
Could you please elaborate more?

Reply:
if TraceFlags is being converted anyway to '1' or '0' wouldn't be easier if they were declared like this instead of `0` or `0x1 << 0` ?

Reply:
I know we have defined this as bitmap, it just looks weird that instead of using enum, we use `'0'` directly, maybe we should then have some new enum defined `TraceFlagsString` `TraceFlagsGRPC` etc. and then simply us that ?

Reply:
I would like this to get merge for now and move this discussion in separate issue to unblock the bug and `0.7.0` release.

Reply:
@mayurkale22 fine for me